bitcoin/src/consensus/encode.rs

```diff
@@ -15,9 +15,38 @@
 //! typically big-endian decimals, etc.)
 //!
 
+//#[macro_export]
+macro_rules! impl_decodable_using_decode {
+    ($type:ty) => {
+        impl crate::consensus::encode::Decodable for $type {
+            #[inline]
+            fn consensus_decode_from_finite_reader<R: crate::io::BufRead + ?Sized>(
+                reader: &mut R,
+            ) -> core::result::Result<Self, crate::consensus::encode::Error> {
+                crate::consensus::encode::decode_sync::<<$type as consensus_encoding::Decode>::Decoder, _>(reader).map_err(Into::into)
+            }
+        }
+    }
+}
+pub(crate) use impl_decodable_using_decode;
+
+macro_rules! impl_encodable_using_encode {
+    ($type:ty) => {
+        impl crate::consensus::encode::Encodable for $type {
+            fn consensus_encode<W: Write + ?Sized>(&self, writer: &mut W) -> core::result::Result<usize, io::Error> {
+                crate::consensus::encode::encode_sync(self, writer)
+            }
+        }
+    }
+}
+
+pub(crate) use impl_encodable_using_encode;
+
```

Comment:
These are just glue, right? The `Encodable` trait goes eventually  once `consenus-encoding` is released?

Reply:
Yes, for compatibility, to be able to upgrade progressively (and thus test the code) without breaking anything.